tests/load/locustfile.py

```diff
@@ -96,23 +151,57 @@ def test_basic(self):
 
     @task
     def test_basic_topic(self):
+        """ Test a basic message transaction using a "topic".
+
+        "Topic" messages will replace prior, queued instances. A topic can be
+        any UA defined, URL Safe base64 compliant string. Upon reconnection,
+        a UA should only get one of each "topic" message that contains only the
+        latest sent data.
+
+        Topic messages are not terribly common, only about 10% of incoming messages
+        use topics.
+        """
+
+        # A Channel ID is how the client User Agent differentiates between various
+        # Web App push notification recipients.
         channel_id = str(uuid.uuid4())
+
+        # Create a connection to the Autoconnect server.
         with closing (
             create_connection(
             self.user.environment.parsed_options.websocket_url,
             header={"Origin": "http://localhost:1337"},
             ssl=False,
             )
         ) as ws:
+            # Connections must say hello after connecting to the server, otherwise
+            # the connection is quickly dropped.
             body = json.dumps(dict(messageType="hello", use_webpush=True))
             ws.send(body)
+
+            # The "hello" response also contains the UserAgent ID (UAID) for the
+            # user agent. The value is random and will be reassigned on reconnection
+            # for passive connections. This value is finalized when an connection
+            # becomes active.
             res = json.loads(ws.recv())
             assert res["messageType"] == "hello"
             uaid = res["uaid"]
+
+            # Register an endpoint for delivery of the message.
             body = json.dumps(dict(messageType="register", channelID=channel_id))
             ws.send(body)
             res = json.loads(ws.recv())
 
+            # NOTE: We are disconnecting from the Autoconnect server so that
+            # we can potentially send multiple topic messages. If we were still
+            # connected the server would deliver the messages as it received them.
+            # That is to be expected.
+
+        # NOTE: I believe that this is cruft from an earlier system. This condition
+        # should just be replaced with
+        # ```
+        # endpoint_url = res["pushEndpoint"]
+        # ```
```

Comment:
Do you think we need these notes here? Or would it be better off to track these externally so when we fix them, we won't have to edit this file again.

Reply:
I'm fine either way. I wanted to capture things like this in the code for now since I was going through it.

Reply:
I had a chat with Kat, I think I didn't understand the context of this PR. Thank you for going through and explaining all the steps, I learned some stuff for sure!